gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/materializer/HiveMaterializer.java

```diff
@@ -0,0 +1,155 @@
+   * @param dataset {@link HiveDataset} for the source table.
+   * @param destinationTable {@link StageableTableMetadata} specifying staging and target tables metadata.
+   */
+  public static HiveWorkUnit tableCopyWorkUnit(HiveDataset dataset, StageableTableMetadata destinationTable,
+      @Nullable String partitionName) {
+    HiveWorkUnit workUnit = new HiveWorkUnit(dataset);
+    workUnit.setProp(MATERIALIZER_MODE_KEY, MaterializerMode.TABLE_COPY.name());
+    workUnit.setProp(STAGEABLE_TABLE_METADATA_KEY, HiveSource.GENERICS_AWARE_GSON.toJson(destinationTable));
+    if (!Strings.isNullOrEmpty(partitionName)) {
+      workUnit.setPartitionName(partitionName);
+    }
+    TaskUtils.setTaskFactoryClass(workUnit, HiveMaterializerTaskFactory.class);
+    return workUnit;
+  }
+
+  /**
+   * Create a work unit to materialize a table / view to a target table using a staging table in between.
+   * @param dataset {@link HiveDataset} for the source table.
+   * @param storageFormat format in which target table should be written.
+   * @param destinationTable {@link StageableTableMetadata} specifying staging and target tables metadata.
+   */
+  public static HiveWorkUnit viewMaterializationWorkUnit(HiveDataset dataset, HiveConverterUtils.StorageFormat storageFormat,
+      StageableTableMetadata destinationTable, @Nullable String partitionName) {
+    HiveWorkUnit workUnit = new HiveWorkUnit(dataset);
+    workUnit.setProp(MATERIALIZER_MODE_KEY, MaterializerMode.TABLE_MATERIALIZATION.name());
+    workUnit.setProp(STORAGE_FORMAT_KEY, storageFormat.name());
+    workUnit.setProp(STAGEABLE_TABLE_METADATA_KEY, HiveSource.GENERICS_AWARE_GSON.toJson(destinationTable));
+    if (!Strings.isNullOrEmpty(partitionName)) {
+      workUnit.setPartitionName(partitionName);
+    }
+    TaskUtils.setTaskFactoryClass(workUnit, HiveMaterializerTaskFactory.class);
+    return workUnit;
+  }
+
+  /**
+   * Create a work unit to materialize a query to a target table using a staging table in between.
+   * @param query the query to materialize.
+   * @param storageFormat format in which target table should be written.
+   * @param destinationTable {@link StageableTableMetadata} specifying staging and target tables metadata.
+   */
+  public static WorkUnit queryMaterializationWorkUnit(String query, HiveConverterUtils.StorageFormat storageFormat,
+      StageableTableMetadata destinationTable) {
+    WorkUnit workUnit = new WorkUnit();
+    workUnit.setProp(MATERIALIZER_MODE_KEY, MaterializerMode.QUERY_MATERIALIZATION.name());
+    workUnit.setProp(STORAGE_FORMAT_KEY, storageFormat.name());
+    workUnit.setProp(QUERY_TO_MATERIALIZE_KEY, query);
+    workUnit.setProp(STAGEABLE_TABLE_METADATA_KEY, HiveSource.GENERICS_AWARE_GSON.toJson(destinationTable));
+    TaskUtils.setTaskFactoryClass(workUnit, HiveMaterializerTaskFactory.class);
+    HiveTask.disableHiveWatermarker(workUnit);
```

Comment:
This disabling of watermark seems confusing here. 
In the `HiveMaterializerSource` where this method is actually invoked, there's also disabling there, and that disabling applies to all three kinds of materialization. 

Can you also add more docs explaining the necessity of disabling watermark?

Reply:
+1 would want to understand this better

Reply:
Watermarker needs to be disabled for query result materialization because there is no such thing as a modtime for a query. The other materializations don't disable it.

In the source it is disabled simply because it is an example source, and it simplifies usage (no state store required, views may not have a watermark, etc.)